sdk/core/azure_core_amqp/src/messaging.rs

```diff
@@ -12,6 +12,16 @@ pub enum TerminusDurability {
     UnsettledState,
 }
 
+impl From<TerminusDurability> for AmqpSymbol {
+    fn from(durability: TerminusDurability) -> Self {
+        match durability {
+            TerminusDurability::None => "none".into(),
```

Comment:
Nit: I'm assuming this then calls into code that converts a `&str` into an `AmqpSymbol::String` or something like that? I would skip the automagical conversion here and be explicit. One would hope the compiler or linker would be smart enough to elide the extra hops, but seems more efficient to go straight into the appropriate variant.

Reply:
That makes a ton of sense. Fixing.